bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapseHandler.js

```diff
@@ -82,16 +85,82 @@ class ViewHandler extends StateHandler {
         setTimeout(() => this.sandbox.postRequestByName('RemoveMapLayerRequest', [id]), ANIMATION_TIMEOUT);
     }
 
+    // Check if we need to show confirm dialog when activating a whole group
+    showWarn (group) {
+        if (group.layers.length > 9) {
+            if (!group.id || this.state.showWarn.includes(group.id)) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn, group.id];
+            // set groups id to showWarn array so we show the confirm dialog
+            this.updateState({ showWarn });
+        } else {
+            return false;
+        }
+    }
+
+    deactivateGroup (group) {
+        if (this.state.showWarn.includes(group.id)) {
+            const index = this.state.showWarn.indexOf(group.id);
+            if (index === -1) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn];
+            showWarn.splice(index, 1);
+            this.updateState({ showWarn });
+        }
+    }
+
+    // active all layers in selected group
+    addGroup (group) {
+        for (var i in group.layers) {
+            if (!group.layers[i]._id || this.state.selectedLayerIds.includes(group.layers[i]._id)) {
+                continue;
+            }
+            const selectedLayerIds = [...this.state.selectedLayerIds, group.layers[i]._id];
+            this.updateState({ selectedLayerIds });
+            setTimeout(() => this.sandbox.postRequestByName('AddMapLayerRequest', [group.layers[i]._id]), ANIMATION_TIMEOUT);
+        }
+        const index = this.state.showWarn.indexOf(group.id);
+        if (index === -1) {
+            return;
+        }
+        const showWarn = [...this.state.showWarn];
+        showWarn.splice(index, 1);
+        this.updateState({ showWarn });
+    }
+
+    // deactivate all layers in group
+    removeGroup (group) {
+        for (var i in group.layers) {
+            const index = this.state.selectedLayerIds.indexOf(group.layers[i]._id);
+            if (index === -1) {
+                continue;
+            }
+            const selectedLayerIds = [...this.state.selectedLayerIds];
+            selectedLayerIds.splice(index, 1);
+            this.updateState({ selectedLayerIds });
+            setTimeout(() => this.sandbox.postRequestByName('RemoveMapLayerRequest', [group.layers[i]._id]), ANIMATION_TIMEOUT);
```

Comment:
Do you need to modify the state in addition to sending the request? The request should trigger AfterMapLayerRemoveEvent that should update the state.

Reply:
Maybe gather the layer ids that are removed and and sent them after single timeout instead of generating multiple timeouts in a loop?

Reply:
Not completely sure how this request works, this was how it was before the changes minus the loop. Can we send an array of the removed "selectedLayerIds" with the request (so have the loop stay the same but gather ids in an array and move the request out of the loop) or how should this be done?

Reply:
You need to send multiple requests. I'm just thinking that they could probably send them in a single setTimeout() instead of generating multiple calls to setTimeout()

Reply:
Not sure how to implement that, could you suggest one? Researching for loops with setTimeout I only encountered a solution that mentioned to switch var to let, I made that change.